src/App.tsx

```diff
@@ -1,24 +1,252 @@
+import classNames from 'classnames';
 import { UserWarning } from './UserWarning';
+import { Todo } from './types/Todo';
+import { ErrorType } from './enums/Errors';
+import {
+  getTodos, createTodo, deleteTodo, patchTodo,
+} from './api/todos';
+import { TodoList } from './components/TodoList/TodoList';
+import { Loader } from './components/Loader/Loader';
+import { TodoFilter } from './components/Todo/TodoFilter';
+import { FilterType } from './enums/FilterType';
+import { getFilteredTodos } from './helpers/GetTodosFiltered';
 
-const USER_ID = 0;
+const USER_ID = 6981;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isListLoading, setListLoading] = useState(false);
+  const [filterType, setFilterType] = useState(FilterType.All);
+  const [activeTodos, completedTodosCount] = useMemo(
+    () => [
+      todos.filter(({ completed }) => !completed).length,
+      todos.filter(({ completed }) => completed).length,
+    ],
+    [todos],
+  );
+  const [errorMessage, setErrorMessage] = useState('');
+  const [query, setQuery] = useState('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [disabledInput, setDisabledInput] = useState(false);
+  const [loadingIds, setLoadingIds] = useState<number[]>([]);
+
+  const closeError = () => {
+    setErrorMessage('');
+  };
+
+  async function getTodosFromServer() {
+    setListLoading(true);
+    try {
+      const todosFromServer = await getTodos(USER_ID);
+
+      setTodos(todosFromServer);
+    } catch (error) {
+      setErrorMessage(ErrorType.UnableToLoad);
+    }
+
+    setListLoading(false);
+  }
+
+  useEffect(() => {
+    setErrorMessage('');
+    getTodosFromServer();
+    setTimeout(() => setErrorMessage(''), 3000);
+  }, []);
+
+  const visibleTodos = getFilteredTodos(todos, filterType);
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
+  const addTodo = async (title: string) => {
+    setDisabledInput(true);
+
+    const newTodo = {
+      title,
+      userId: USER_ID,
+      completed: false,
+    };
+
+    if (query.trim() === '') {
+      Notify
+        .failure(ErrorType.EmptyQuery);
+      setDisabledInput(false);
+      setQuery('');
+    } else {
+      setTempTodo({ ...newTodo, id: 0 });
+
+      try {
+        const result = await createTodo(newTodo);
+
+        setTodos((state) => [...state, result] as Todo[]);
+      } catch (error) {
+        setErrorMessage(ErrorType.UnableToAdd);
+      } finally {
+        setDisabledInput(false);
+        setTempTodo(null);
+        setQuery('');
+      }
+    }
+  };
+
+  const removeTodo = async (id: number) => {
+    try {
+      setLoadingIds(state => [...state, id]);
+      await deleteTodo(id);
+      setTodos(todos.filter(todo => todo.id !== id));
+    } catch (error) {
+      setErrorMessage(ErrorType.UnableToRemove);
+      setTimeout(() => {
+        setErrorMessage('');
+      }, 3000);
+    } finally {
+      setLoadingIds([]);
```

Comment:
But what if there will be many loading ids? You adding one id to existing array, so it's logical to filter this array to remove one id instead of setting empty array here